docs-ref-conceptual/create-an-azure-service-principal-azure-cli.md

```diff
@@ -36,58 +36,79 @@ The `appId` and `tenant` keys appear in the output of `az ad sp create-for-rbac`
 
-Without any authentication parameters, password-based authentication is used with a random password created for you.
+With password-based authentication, a random password is created for you.  Along with a `--name`, you must specify a `--scope` as this value does not have a default.  If you prefer, you can set the role assignment later by using [az role assignment create](/cli/azure/role/assignment#az-role-assignment-create).
+
+```azurecli-interactive
+az ad sp create-for-rbac --name myServicePrincipalName \
+                         --role reader \
+                         --scope /subscriptions/mySubscriptionID/resourceGroups/myResourceGroupName
+```
 
-  ```azurecli-interactive
-  az ad sp create-for-rbac --name ServicePrincipalName --role Contributor
-  ```
+You can also create a service principal using variables.
 
-> [!IMPORTANT]
-> As of Azure CLI 2.0.68, the `--password` parameter to create a service principal with a user-defined password
-> is __no longer supported__ to prevent the accidental use of weak passwords.
+```azurecli-interactive
+let "randomIdentifier=$RANDOM*$RANDOM"  
+servicePrincipalName="msdocs-sp-$randomIdentifier"
+roleName="azureRoleName"
+subscriptionID=$(az account show --query id -o tsv)
+# Verify the ID of the active subscription
+echo "Using subscription ID $subscriptionID"
+resourceGroup="myResourceGroupName"
+
+echo "Creating SP for RBAC with name $servicePrincipalName, with role $roleName and in scope /subscriptions/$subscriptionID/resourceGroups/$resourceGroup"
+az ad sp create-for-rbac --name $servicePrincipalName --role $roleName --scope /subscriptions/$subscriptionID/resourceGroups/$resourceGroup
+```
 
-The output for a service principal with password authentication includes the `password` key. __Make sure__ you copy this value - it can't be retrieved. If you forget the password, [reset the service principal credentials](#6-reset-credentials).
+The output for a service principal with password authentication includes the `password` key. __Make sure you copy this value__ - it can't be retrieved. If you lose the password, [reset the service principal credentials](#6-reset-credentials).
 
 ### Certificate-based authentication
 
-For certificate-based authentication, use the `--cert` argument. This argument requires that you hold an existing certificate. Make sure any tool that uses this service principal has access to the certificate's private key. Certificates should be in an ASCII format such as PEM, CER, or DER. Pass the certificate as a string, or use the `@path` format to load the certificate from a file.
+For certificate-based authentication, use the `--cert` parameter. This parameter requires that you hold an existing certificate. Make sure any tool that uses this service principal has access to the certificate's private key. Certificates should be in an ASCII format such as PEM, CER, or DER. Pass the certificate as a string, or use the `@path` format to load the certificate from a file.
```

Comment:
In CLI's in-tool help, we actually use the word 'argument':

```
> az ad sp create-for-rbac -h
...
Arguments
    --name -n               : Display name of the service principal. If not present, default to
                              azure-cli-%Y-%m-%d-%H-%M-%S where the suffix is the time of creation.
```

Reply:
> Is there a difference between an "argument" and a "parameter"?

In computer science, there is. **parameter** is the name, while **argument** is the value.

But in Azure CLI context, they are the same and can be used interchangeably.